ThreeDSecure/src/main/java/com/braintreepayments/api/threedsecure/ThreeDSecureLookup.kt

```diff
@@ -0,0 +1,59 @@
+package com.braintreepayments.api.threedsecure
+
+import android.os.Parcelable
+import com.braintreepayments.api.sharedutils.Json
+import kotlinx.parcelize.Parcelize
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Data class to parse and contain 3D Secure lookup params
+ */
+@Parcelize
+data class ThreeDSecureLookup internal constructor(
+    val acsUrl: String? = null,
+    val md: String = "",
+    val termUrl: String = "",
+    val pareq: String = "",
+    val threeDSecureVersion: String = "",
+    val transactionId: String = ""
+) : Parcelable {
+
+    companion object {
+        private const val ACS_URL_KEY = "acsUrl"
+        private const val MD_KEY = "md"
+        private const val TERM_URL_KEY = "termUrl"
+        private const val PA_REQ_KEY = "pareq"
+        private const val THREE_D_SECURE_VERSION_KEY = "threeDSecureVersion"
+        private const val TRANSACTION_ID_KEY = "transactionId"
+
+        /**
+         * Used to parse a response from the Braintree Gateway to be used for 3D Secure.
+         *
+         * @param jsonString The json response from the Braintree Gateway 3D Secure lookup route.
+         * @return The [ThreeDSecureLookup] to use when performing 3D Secure authentication.
+         * @throws JSONException when parsing fails.
+         */
+        @JvmStatic
+        @Throws(JSONException::class)
+        fun fromJson(jsonString: String): ThreeDSecureLookup {
+            val json = JSONObject(jsonString)
+            return ThreeDSecureLookup(
+                acsUrl = Json.optString(json, ACS_URL_KEY, null),
+                md = Json.optString(json, MD_KEY, ""),
+                termUrl = Json.optString(json, TERM_URL_KEY, ""),
```

Comment:
I notice these changing from get to opt - are we ok with losing the JSONException when they are missing? I don't know if they ever should be and assume an error would surface down the line instead?

Reply:
I updated from get to opt to be consistent with all the properties. Hmm yeah something would probably fail further in the flow, but since these fields are all non-null, I think failing fast here makes the most sense. I'll update these all from opt to get.

Reply:
I ended up leaving optString for `acsUrl` since it's the only nullable field. The others have been updated to use `getString`.

Reply:
It looks like the previous behavior is that the pareq value is set to an empty string if it's missing from the response. Do you know if the `pareq` value is required for the 3DS flow? If it is, are we still good with throwing an exception if it's missing?

Unit test that checks that `pareq` is an empty string when it's missing from the response.
```java
    @Test
    public void fromJson_whenPareqMissing_parsesCorrectly() throws JSONException {
        JSONObject lookupV2MissingPareq = new JSONObject(
                Fixtures.THREE_D_SECURE_V2_LOOKUP_RESPONSE_MISSING_PAREQ).getJSONObject("lookup");
        ThreeDSecureLookup sut = ThreeDSecureLookup.fromJson(lookupV2MissingPareq.toString());
        assertEquals("", sut.getPareq());
    }
```

Reply:
I ended up keeping the existing behavior. If the `pareq` value from the response is missing or null, an empty string is set.

Reply:
Going to merge the PR, but open to updating this is folks feel that we should handle the null case differently.